cosign/README.md

```diff
@@ -0,0 +1,79 @@
+# Cosign Verifier
+
+## Introduction
+
+This package contains the source code for the ratify-verifier-go cosign client library.
+
+This package is built on the `sigstore-go` library, a low-level Go library that provides a suite of tools and standards designed for signing, verifying, and securing software supply chains.
+`sigstore-go` focuses on the core functionalities of signing and verification, offering a minimal and user-friendly API without integration of registry operations or additional features like container image management.
+In contrast, Cosign contains higher-level APIs within the Sigstore ecosystem that integrates these functionalities with container registries and other tools for end-to-end supply chain security.
+Cosign builds on `sigstore-go` and extends its capabilities to include container image signing, verification, and storage in registries.
+
+Using `sigstore-go` library makes it a more streamlined choice for developers seeking to implement lightweight and flexible signature verification.
+Additionally, the Sigstore team plans to refactor parts of Cosign to rely more heavily on `sigstore-go`, making the interaction between the two libraries even closer.
+
+## Concepts
+
+Artifact Signatures  
+
+- Artifacts may be signed using different cryptographic schemes. The verifier must support a variety of signature formats, including RSA, ECDSA, and Ed25519.
+- The use of DSSE helps bundle signatures with metadata, providing additional context and validation capabilities.
+
+Transparency Logs  
+
+- Transparency logs, such as those maintained by the Sigstore project, provide an immutable record of signature entries.
+- Verifying against these logs adds an extra layer of assurance that a signature has been publicly disclosed and has not been secretly revoked or altered.
+- As a primary and recommanded solution, Cosign will then store the signature and certificate in the Rekor transparency log.
+
+Verification Workflows  
+
+- The verifier must account for different workflows, such as verifying container images, software binaries, or other artifacts.
+- Each workflow may involve different states of metadata, such as embedded signatures versus detached signatures, and requires customization accordingly.
+- `sigstore-go` library includes a few abstractions to support different use cases, testing, and extensibility:
+
+  - `SignedEntity` - an interface type respresenting a signed message or attestation, with a signature and metadata, implemented by `Bundle`, a type which wraps the `Bundle` type from `protobuf-specs`.
+  - `TrustedMaterial` - an interface type representing a trusted set of keys or certificates for verifying certificates, timestamps, and artifact transparency logs, implemented by `TrustedRoot`
+
+## Scenarios
```

Comment:
In this section, we only list those scenarios we support.

Reply:
For each scenario, you should also mention what's the input and potential output.

Reply:
As discussed, will provide general I/O in shape of payload, sig, trust content -> outcome, error